bin/celeste_mcmc.jl

```diff
@@ -0,0 +1,166 @@
+                                   neighbors::Vector{CatalogEntry}, 
+                                   images::Vector{TiledImage})
+    # preprocssing
+    cat_local = vcat(entry, neighbors)
+    vp = Vector{Float64}[init_source(ce) for ce in cat_local]
+    patches, tile_source_map = Infer.get_tile_source_map(images, cat_local)
+    ea = ElboArgs(images, vp, tile_source_map, patches, [1])
+    Infer.fit_object_psfs!(ea, ea.active_sources)
+    Infer.trim_source_tiles!(ea)
+    active_pixels = get_active_pixels(ea)
+
+    # generate the star logpdf and initial MCMC chain state
+    # states are packed together [lnr, cu, cg, cr, ci, ra, dec]
+    # TODO create a sample_from_star_prior function (over dispersion...)
+    star_logpdf, star_logprior =
+        Model.make_star_logpdf(images, active_pixels, ea)
+    star_state       = Model.init_star_state(entry)
+    star_param_names = ["lnr", "cug", "cgr", "cri", "ciz", "ra", "dec"]
+    star_sim = run_slice_sampler(star_logpdf, star_state, 200, star_param_names)
+
+    # generate the galaxy logpdf and initial galaxy MCMC chain state
+    # TODO create a sample_from_gal_prior function (with over dispersion)
+    gal_logpdf, gal_logprior =
+        Model.make_galaxy_logpdf(images, active_pixels, ea)
+    gal_state   = Model.init_galaxy_state(entry)
+    gal_param_names = [star_param_names;
+                       ["gdev", "gaxis", "gangle", "gscale"]]
+    gal_sim = run_slice_sampler(gal_logpdf, gal_state, 200, gal_param_names)
+
+    # generate pointers to star/gal type (to infer p(star | data))
+    #type_samples, type_lls = run_star_gal_switcher(star_lls, gal_lls)
+
+    # return all samples
+    #return SampleResult(star_samples, gal_samples, type_samples,
+    #                    star_lls, gal_lls, type_lls)
+end
+
+
+"""
+Run a slice sampler for N steps
+"""
+function run_slice_sampler(lnpdf::Function,
+                           th0::Vector{Float64},
+                           N::Int,
+                           param_names::Vector{String})
+    # slice sample as in example:
+    # http://mambajl.readthedocs.io/en/latest/examples/line_amwg_slice.html
+    sim = Mamba.Chains(N, length(th0), names = param_names)
+    th  = Mamba.SliceUnivariate(th0, 1., lnpdf)
+    for i in 1:N
+      if mod(i, 25) == 0
+          println("   sample ", i)
+      end
+      sample!(th)
+      sim[i, :, 1] = th
+    end
+    Mamba.describe(sim)
+    return sim
+end
+
+
+function run_star_gal_switcher(star_lls::Vector{Float64},
+                               gal_lls::Vector{Float64})
+    #TODO switch between states of the chain, alt to RJMCMC
+end
+
+
+function one_node_infer_mcmc(rcfs::Vector{RunCamcolField},
+                             stagedir::String;
+                             objid="",
+                             box=BoundingBox(-1000., 1000., -1000., 1000.),
+                             primary_initialization=true)
+    # catalog
+    duplicate_policy = primary_initialization ? :primary : :first
```

Comment:
I agree, and probably `api.jl` should call the mcmc module rather than the other way around, to avoid spaghetti code.